llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7539,6 +7539,7 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
                         CM.CostKind);
   precomputeCosts(BestPlan, BestFactor.Width, CostCtx);
   assert((BestFactor.Width == LegacyVF.Width ||
+          Legal->hasUncountableEarlyExit() ||
```

Comment:
Ideally we would be able to reliably check if a VPlan has an early exit directly on the VPlan instead of relying on Legal to do so. Could we check if we have either multiple exit blocks or a single exit block with multiple predecessors?